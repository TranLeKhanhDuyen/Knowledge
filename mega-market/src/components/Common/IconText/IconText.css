.icontext-container {
  display: flex;
  align-items: center;
  border: none;
  background-color: transparent;
  cursor: pointer;
  gap: 6px;
  width: max-content;
}

.icontext-container-right {
  flex-direction: row-reverse;
}

/* SIZE */
.icontext-title-sm {
  font-size: var(--font-sm);
  line-height: 18px;
  font-weight: 400;
}

.icontext-title-md {
  font-size: var(--font-md);
}

.img-icontext-sm {
  width: 18px;
  height: 18px;
}

.img-icontext-md {
  width: 24px;
  height: 24px;
}

/* TYPE */
.icontext-content-primary {
  display: flex;
  gap: 40x ;
}

.icontext-content-secondary {
  display: flex;
  flex-direction: column;
  gap: 2px
}

.icontext-title-primary {
  color: var(--text-secondary-color);
}

.icontext-sub-title-primary {
  font-weight: var(--fw-bold);
  color: var(--text-secondary-color);
}

.icontext-title-secondary {
  color: var(--white-color);
}

.icontext-sub-title-secondary {
  color: var(--white-color);
}
