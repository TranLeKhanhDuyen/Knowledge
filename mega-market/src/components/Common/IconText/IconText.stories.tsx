import type { Meta, StoryObj } from '@storybook/react'

import { IconText } from '@components'

export default {
  title: 'Components/IconText',
  component: IconText,
  tags: ['autodocs'],
  argTypes: {
    size: { description: 'The size of button' },
    additionalClass: { description: 'Add class to the button' },
    title: { description: 'The title of button.' },
    subTitle: { description: 'The subs title of button.' },
    iconPosition: {
      description: 'The position of the icon relative to the text',
      control: {
        type: 'radio',
        options: ['left', 'right']
      }
    }
  }
} as Meta
type Story = StoryObj<typeof IconText>
export const Default: Story = {
  args: {
    size: 'sm',
    icon: 'location',
    title: 'Deliver to',
    subTitle: '423651'
  }
}
export const IconLeft: Story = {
  ...Default,
  args: {
    icon: 'truck',
    title: 'Track your order',
    iconPosition: 'left'
  }
}
export const IconRight: Story = {
  ...Default,
  args: {
    icon: 'buy',
    title: 'Groceries',
    iconPosition: 'right',
    additionalClass: 'select'
  }
}
