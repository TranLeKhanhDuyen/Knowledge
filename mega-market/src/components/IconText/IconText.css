.icontext-container {
  display: flex;
  align-items: center;
  border: none;
  background-color: transparent;
  cursor: pointer;
  gap: 6px;
  width: max-content;
}

.icontext-container-right {
  flex-direction: row-reverse;
}

.select.icontext-container {
  background-color: var(--tertiary-color);
  padding: 9px 14px;
  border-radius: 18px;
}

.select .icontext-title {
  color: var(--black-color);
}

.icontext-title {
  color: var(--text-secondary-color);
}

.icontext-title-sm {
  font-size: var(--font-sm);
  line-height: 18px;
  font-weight: 400;
}

.icontext-title-md {
  font-size: var(--font-md);
}

.icontext-sub-title {
  font-weight: var(--fw-bold);
}

.img-icontext-sm {
  width: 18px;
  height: 18px;
}

.img-icontext-md {
  width: 24px;
  height: 24px;
}
