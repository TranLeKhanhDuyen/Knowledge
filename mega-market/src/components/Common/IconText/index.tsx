import IconSvg, {IconSvgProps } from '@components/Common/IconSvg'
import './IconText.css'

type TIconTextSize = 'sm' | 'md' | 'lg'
type TIconPosition = 'left' | 'right'
type TIconTextVariant = 'primary' | 'secondary'

export interface IIconTextProps
  extends React.HTMLAttributes<HTMLParagraphElement> {
  size?: TIconTextSize
  iconPosition?: TIconPosition
  title?: string
  subTitle?: string
  icon: IconSvgProps['name']
  additionalClass?: string
  type?: TIconTextVariant
}

const IconText: React.FC<IIconTextProps> = ({
  size = 'sm',
  iconPosition = 'left',
  title,
  subTitle,
  icon,
  additionalClass = '',
  children,
  type = 'primary',
  ...props
}) => {
  const containerClass = `icontext-container icontext-container-${size} ${
    iconPosition === 'right' ? 'icontext-container-right' : ''
  } ${additionalClass ?? ''}`
  const iconClass = `img-icontext-${size} img-icontext-${type}`
  const titleClass = `icontext-title icontext-title-${size} icontext-title-${type}`
  const subTitleClass = `icontext-title icontext-sub-title-${type}`

  return (
    <article className={containerClass} {...props}>
      <IconSvg name={icon} className={iconClass} />
      <aside className={`icontext-content icontext-content-${type}`}>
        <p className={titleClass}>{title}</p>
        <p className={subTitleClass}>{subTitle}</p>
        {children}
      </aside>
    </article>
  )
}

export default IconText
