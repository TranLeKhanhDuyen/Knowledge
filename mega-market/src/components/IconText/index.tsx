import IconSvg, { IconSvgProps } from '@components/IconSvg'
import '@components/IconText/IconText.css'

type TIconTextSize = 'sm' | 'md' | 'lg'
type TIconPosition = 'left' | 'right'

export interface IIconTextProps
  extends React.HTMLAttributes<HTMLParagraphElement> {
  size?: TIconTextSize
  iconPosition?: TIconPosition
  title?: string
  subTitle?: string
  icon: IconSvgProps['name']
  additionalClass?: string
}

const IconText: React.FC<IIconTextProps> = ({
  size = 'sm',
  iconPosition = 'left',
  title,
  subTitle,
  icon,
  additionalClass,
  children,
  ...props
}) => {
  const containerClass = `icontext-container icontext-container-${size} ${
    iconPosition === 'right' ? 'icontext-container-right' : ''
  } ${additionalClass ?? ''}`
  const iconClass = `img-icontext-${size}`
  const titleClass = `icontext-title icontext-title-${size} ${
    additionalClass ?? ''
  }`
  const subTitleClass = `icontext-title icontext-sub-title ${
    additionalClass ?? ''
  }`

  return (
    <p className={containerClass} {...props}>
      <IconSvg name={icon} className={iconClass} />
      {title && <span className={titleClass}>{title}</span>}
      {subTitle && <span className={subTitleClass}>{subTitle}</span>}
      {children}
    </p>
  )
}

export default IconText
