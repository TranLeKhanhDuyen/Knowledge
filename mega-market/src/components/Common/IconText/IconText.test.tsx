import { render } from '@testing-library/react'
import IconText, { IIconTextProps } from './index'

// Mock IconSvg component
jest.mock('@components/Common/IconSvg', () => ({ name }: { name: string }) => (
  <svg data-testid={`icon-${name}`} />
))

describe('IconText', () => {
  const renderIconText = (props: Partial<IIconTextProps> = {}) => {
    const defaultProps: IIconTextProps = {
      size: 'sm',
      iconPosition: 'left',
      title: 'Default Title',
      subTitle: 'Default SubTitle',
      icon: 'search',
      additionalClass: '',
      type: 'primary'
    }
    return render(<IconText {...defaultProps} {...props} />)
  }

  test('renders with the correct size class', () => {
    const { container } = renderIconText({ size: 'lg' })
    const articleElement = container.querySelector('article')
    expect(articleElement).toHaveClass('icontext-container-lg')
  })

  test('renders with the correct icon position class', () => {
    const { container } = renderIconText({ iconPosition: 'right' })
    const articleElement = container.querySelector('article')
    expect(articleElement).toHaveClass('icontext-container-right')
  })

  test('renders the correct icon', () => {
    const { getByTestId } = renderIconText({ icon: 'user' })
    expect(getByTestId('icon-user')).toBeInTheDocument()
  })

  test('applies additional class names', () => {
    const { container } = renderIconText({ additionalClass: 'custom-class' })
    const articleElement = container.querySelector('article')
    expect(articleElement).toHaveClass('custom-class')
  })

  test('renders the correct title and subtitle', () => {
    const { getByText } = renderIconText({
      title: 'Test Title',
      subTitle: 'Test SubTitle'
    })
    expect(getByText('Test Title')).toBeInTheDocument()
    expect(getByText('Test SubTitle')).toBeInTheDocument()
  })

  test('renders children correctly', () => {
    const { getByText } = renderIconText({
      children: <span>Child Content</span>
    })
    expect(getByText('Child Content')).toBeInTheDocument()
  })

  test('matches snapshot', () => {
    const { asFragment } = renderIconText()
    expect(asFragment()).toMatchSnapshot()
  })
})
