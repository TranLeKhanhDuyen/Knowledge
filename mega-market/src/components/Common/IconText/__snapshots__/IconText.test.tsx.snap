// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconText matches snapshot 1`] = `
<DocumentFragment>
  <article
    class="icontext-container icontext-container-sm  "
  >
    <svg
      data-testid="icon-search"
    />
    <aside
      class="icontext-content icontext-content-primary"
    >
      <p
        class="icontext-title icontext-title-sm icontext-title-primary"
      >
        Default Title
      </p>
      <p
        class="icontext-title icontext-sub-title-primary"
      >
        Default SubTitle
      </p>
    </aside>
  </article>
</DocumentFragment>
`;
